src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp

```diff
@@ -438,7 +454,64 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         return RS_(L"Profile_AnswerbackMessageNone");
     }
 
-    Editor::AppearanceViewModel ProfileViewModel::DefaultAppearance()
+    Windows::UI::Color ProfileViewModel::TabColorPreview() const
+    {
+        if (const auto modelVal = _profile.TabColor())
+        {
+            // user defined an override value
+            return Windows::UI::Color{
+                .A = 255,
+                .R = modelVal.Value().R,
+                .G = modelVal.Value().G,
+                .B = modelVal.Value().B
+            };
+        }
+        // set to null --> deduce value from theme
+        return TabThemeColorPreview();
+    }
+
+    Windows::UI::Color ProfileViewModel::TabThemeColorPreview() const
+    {
+        const auto currentTheme = _appSettings.GlobalSettings().CurrentTheme();
+        if (const auto tabTheme = currentTheme.Tab())
+        {
+            // theme.tab.background: theme color must be evaluated
+            if (const auto tabBackground = tabTheme.Background())
+            {
+                const auto& tabBrush = tabBackground.Evaluate(Application::Current().Resources(),
+                                                              Windows::UI::Xaml::Media::SolidColorBrush{ DefaultAppearance().CurrentColorScheme().BackgroundColor().Color() },
+                                                              false);
+                if (const auto& tabColorBrush = tabBrush.try_as<Windows::UI::Xaml::Media::SolidColorBrush>())
+                {
+                    const auto brushColor = tabColorBrush.Color();
+                    return brushColor;
+                }
+            }
+        }
+        else if (const auto windowTheme = currentTheme.Window())
+        {
+            // theme.window.applicationTheme: evaluate light/dark to XAML default tab color
+            // Can also be "Default", in which case we fall through below
+            const auto appTheme = windowTheme.RequestedTheme();
+            if (appTheme == ElementTheme::Dark)
+            {
+                return Windows::UI::ColorHelper::FromArgb(0xFF, 0x28, 0x28, 0x28);

```

Comment:
Why not create the `Color` struct directly? You're doing that above too, after all.